dev/bin/common.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+
+configure_script_dir_path=`dirname "${BASH_SOURCE[0]}"`
+
+# helper functions
+function check_required_variable {
+  var_name="$1"
+  [ -z "$( eval "echo -n \$${var_name}")" ] && die "required environment variable \"${var_name}\" is missing or empty" || true
+}
+
+function die {
+  echo "ERROR: $1"
+  exit
+}
+
+# make sure this script is sourced (no direct execution)
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] || die "script ${BASH_SOURCE[0]} is intended only to be sourced from other scripts"
+
+# list of required tools
+required_tools="
+docker-compose
+jq
+kubectl
+kustomize
+mvn
+ping
+rhoas
+"
+
+# check if required tools are installed
+for tool in $required_tools; do
+  which "$tool" &> /dev/null || die "required tool \"$tool\" is missing"
+done
+
+# useful paths
+root_dir=`realpath ${configure_script_dir_path}/../..`
+dev_dir="${root_dir}/dev"
+local_deploy_dir="${dev_dir}/_deploy"
+dev_config_dir="${dev_dir}/config"
+dev_bin_dir="${dev_dir}/bin"
+dev_logs_dir="${dev_dir}/logs"
+dev_docker_compose_dir="${dev_dir}/docker-compose"
+credentials_dir="${dev_dir}/credentials"
+kustomize_deploy_dir="${local_deploy_dir}/kustomize"
+local_env_file="${local_deploy_dir}/local_env"
+local_config_file="${dev_config_dir}/localconfig"
+kustomize_dir="${root_dir}/kustomize"
+
+fleet_manager_container_name=${FLEET_MANAGER_CONTAINER_NAME:-'openbridge/manager:latest'}
+fleet_shard_container_name=${FLEET_SHARD_CONTAINER_NAME:-'openbridge/shard-operator:latest'}
+executor_container_name=${EXECUTOR_CONTAINER_NAME:-'openbridge/executor:latest'}
+ingress_container_name=${INGRESS_CONTAINER_NAME:-'openbridge/ingress:latest'}
+
+function configure_kafka {
+  retrieve_full_env
+  check_required_variable "MANAGED_KAFKA_INSTANCE_NAME"
+  manager_kafka_credentials_file="${credentials_dir}/${MANAGED_KAFKA_INSTANCE_NAME}.json"
+  admin_sa_name="${MANAGED_KAFKA_INSTANCE_NAME}-admin"
+  admin_sa_credentials_file="${credentials_dir}/${admin_sa_name}.json"
+  ops_sa_name="${MANAGED_KAFKA_INSTANCE_NAME}-ops"
+  ops_sa_credentials_file="${credentials_dir}/${ops_sa_name}.json"
+  mc_sa_name="${MANAGED_KAFKA_INSTANCE_NAME}-mc"
+  mc_sa_credentials_file="${credentials_dir}/${mc_sa_name}.json"
+}
+
+function configure_managed_connectors {
+  retrieve_full_env
+  check_required_variable "OPENSHIFT_OFFLINE_TOKEN"
+  managed_connectors_cluster_id="${MANAGED_CONNECTORS_CLUSTER_ID:-empty}"
+}
+
+function configure_cluster {
+  retrieve_full_env
+  cluster_type=${CLUSTER_TYPE:-minikube}
+  if [ "${cluster_type}" = 'minikube' ]; then
+    configure_minikube
+  elif [ "${cluster_type}" = 'kind' ]; then
+    configure_kind
+  fi
+}
+
+function configure_minikube {
+  retrieve_full_env
+  minikube_profile="${MINIKUBE_PROFILE:-minikube}"
+  minikube_cpus="${MINIKUBE_CPUS:-4}"
+  minikube_memory="${MINIKUBE_MEMORY:-8192}"
+  minikube_kubernetes_version="${MINIKUBE_KUBERNETES_VERSION:-v1.21.6}"
```

Comment:
This should be the same defined in the .github/actions/kind/action.yml 
Can you add a comment to keep them aligned?

Reply:
I was more thinking about an `.env` file in the root of the project. wdyt ?

Reply:
Even better :)

Reply:
implemented. please have a look. I had to update the `kind` github action